### PR TITLE
Adding dist-git-ignore option

### PR DESCRIPTION
**Pull request by adammhaile**

### CODE REVIEW

File: doozerlib/distgit.py [hunk 1]
```diff
@@ -40,6 +40,8 @@
 ---
 """
 
+BASE_IGNORE = [".git", ".oit", ".gitignore", "additional-tags"]
```
Comment (tbielawa) [inline doozerlib/distgit.py]: Comment for this plz: context as to why this variable exists

**Comment by adammhaile** [comment]
*@tbielawa *comment added

File: doozerlib/distgit.py (hunk 1 above)
Comment (tbielawa) [reply doozerlib/distgit.py]: yes

Comment (sosiouxme) [reply doozerlib/distgit.py]: although i might say that we're "ignoring" them in the sense that they don't get cleared out of distgit on rebase, not a rebase ignores source content.

**Comment by adammhaile** [comment]
Sam had a similar content since ignoring implies that we also won't
overwrite... which would also be handy.

**Pull request merged**